doc/specification.md

```diff
@@ -0,0 +1,597 @@
+    , the device should close the `Get` RPC with the generic gRPC
+    [`Status.code`][gRPC status code] set to `UNIMPLEMENTED`.
+
+### 4.2.2 `GetResponse` message
+
+A `GetResponse` contains a list of `AFTEntry` messages. An `AFTEntry` message
+represents an installed gRIBI entry (the data model is defined in
+[section 2](#2-data-model)) and its server acknowledged programming status
+(`rib_status` and `fib_status`). "server acknowledged programming status"
+represents the `AFTResult` at the time of the response being sent to the client
+(regardless if sending the response failed).
+
+*   `rib_status` indicates the programming status of the gRIBI entry in RIB. The
+    value should be either `PROGRAMMED` or `NOT_PROGRAMMED`.
+*   `fib_status` indicates the programming status of the gRIBI entry in FIB.
+    *   When the session parameter is `ack_type` = `RIB_ACK`, it's optional for
+        the device to keep track of FIB programming status of each gRIBI entry.
+        Therefore, this field MAY be set to `UNAVAILABLE`.
+    *   When the session parameter is `ack_type` = `RIB_AND_FIB_ACK`, the value
+        should be either `PROGRAMMED` or `NOT_PROGRAMMED`.
+
+If the specified network instances have no installed gRIBI objects, the device
+should return an empty list of `AFTEntry` and then close the RPC with the
+generic gRPC [`Status.code`][gRPC status code] set to `OK`.
+
+## 4.3 `Flush` RPC
+
+The `Flush` RPC is an unary RPC for clients to remove gRIBI entries from a
+device. A client sends a `FlushRequest` message specifying the target network
+instance where the device should remove all gRIBI entries. The device processes
+the request and responds a `FlushResponse` message indicating the execution
+result.
+
+The `Flush` RPC can be used in some emergency process to get the device out of
+undesirable routing state, therefore:
+
+*   This RPC provides a low complexity method to remove all gRIBI entries in
+    specified network instance.
+*   This RPC allows non primary client (in `SINGLE_PRIMARY` mode) to remove all
+    gRIBI entries in specified network instance.
+
+### 4.3.1 `FlushRequest` Message
+
+A `FlushRequest` message MUST have the `network_instance` populated by client.
+
+*   If `network_instance` is nil or `network_instance.name` is an empty string,
+    the device should reject the request with gRPC error
+    [`Status.code`][gRPC status code] set to `INVALID_ARGUMENT`. The
+    `Status.details` should contain `FlushResponseError` message with `reason`
+    set to `INVALID_NETWORK_INSTANCE`.
+*   If the specified network instance does not exist, the device should reject
+    the request with gRPC error [`Status.code`][gRPC status code] set to
+    `INVALID_ARGUMENT`. The `Status.details` should contain `FlushResponseError`
+    message with `reason` set to `NO_SUCH_NETWORK_INSTANCE`.
+
+#### 4.3.1.1 `election` In `FlushRequest` Message
+
+Only when the client-server is in `SINGLE_PRIMARY` mode (defined in
```

Comment:
Suggest switching this sentence's form around "The client MUST populate the `election` field when the server is in `SINGLE_PRIMARY` mode". I think it specifically the *server* that we need to care about here, because there might be zero active clients at the time of the flush, and I think this is still required.